### ISSUE TITLE
The parameter column_families in the LDBTool::Run function did not take effect.

### ISSUE THREAD

**Issue by dongdongwcpp** [facebook/rocksdb#11581]
The parameter column_families in the LDBTool::Run function did not take effect while using a custom Comparator or other custom scenarios:

``` c++
void LDBTool::Run(int argc, char** argv, Options options,
                  const LDBOptions& ldb_options,
                  const std::vector<ColumnFamilyDescriptor>* column_families) 
LDBTool tool;
tool.Run(argc, argv, rocksdb::Options(), LDBOptions(), &column_families);
```
which seems to be a problem. This issue still exists in the latest version of RocksDB.

### Expected behavior
when using a custom Comparator to open rocksdb,The function `LDBTool::run`  should  set std::vector<ColumnFamilyDescriptor> column_families to ldb tool, and should work fine to admin the database.
### Actual behavior

```
root$ ../build/tools/ldb   --db=./testdb drop_column_family new_cf
Failed: Invalid argument: leveldb.BytewiseComparator: does not match existing comparator my_comparator
```
### Steps to reproduce the behavior
```c++
#include <rocksdb/db.h>
#include <rocksdb/options.h>
#include "rocksdb/ldb_tool.h"

using namespace rocksdb;
class MyComparator : public rocksdb::Comparator {
 public:
  int Compare(const rocksdb::Slice& a, const rocksdb::Slice& b) const override {
    return a.compare(b);
  }
  virtual void FindShortSuccessor(std::string* key) const {}
  virtual void FindShortestSeparator(std::string* start, const Slice& limit) const {}
  const char* Name() const override { return "my_comparator"; }
};

int main(int argc, char** argv) {
  rocksdb::DB* db;
  rocksdb::Options options;
  options.create_if_missing = true;
  options.create_missing_column_families = true;
  rocksdb::ColumnFamilyOptions cf_options;

  cf_options.comparator = new MyComparator;
  // create column family with custom comparator
  std::vector<rocksdb::ColumnFamilyDescriptor> column_families = {
      rocksdb::ColumnFamilyDescriptor(rocksdb::kDefaultColumnFamilyName,cf_options),
      rocksdb::ColumnFamilyDescriptor("new_cf", cf_options),
  };

  if (argc > 2) {
    LDBTool tool;
    tool.Run(argc, argv, rocksdb::Options(), LDBOptions(), &column_families);
    return 0;
  }

  std::vector<ColumnFamilyHandle*> handles;
  rocksdb::DB::Open(options, "./testdb", column_families, &handles, &db);
  db->Put(rocksdb::WriteOptions(), "key1", "value1");
  delete db;
  return 0;
}
```
compile and run:
```shell
g++ --std=c++11 my_comparator.cc -o ldb_test  -I ../include ../build_release/librocksdb.a -lpthread 
#run to create db with two cf
./ldb_test
#run to drop new_cf and got failed
./ldb_test --db=./testdb drop_column_family new_cf
Failed: Invalid argument: leveldb.BytewiseComparator: does not match existing comparator my_comparator
```
### how to fix
set column_families for `command` here:
https://github.com/facebook/rocksdb/blob/15053f3ab47839bc14dc31a8a62df1d23f6babbe/tools/ldb_cmd.cc#L200
I may try to fix it if it's a real issue.

**Comment by ajkr** [comment]
> I may try to fix it if it's a real issue.

Looks like a real issue to me

**Comment by ajkr** [comment]
Fixed by #12261
